string instruction
uint32 time